ddclient.in

```diff
@@ -2155,6 +2168,171 @@ sub geturl {
     return $reply;
 }
 
+######################################################################
+## curl_cmd() function to execute system curl command
+######################################################################
+sub curl_cmd {
+    my @params = @_;
+    my $tmpfile;
+    my $tfh;
+    my $system_curl = subst_var('@syscurlcmd@', '/usr/bin/curl');
+    my %curl_codes = (  ## Subset of error codes from https://curl.haxx.se/docs/manpage.html
+         2 => "Failed to initialize. (Most likely a bug in ddclient, please open issue at https://github.com/ddclient/ddclient)",
+         3 => "URL malformed. The syntax was not correct",
+         5 => "Couldn't resolve proxy. The given proxy host could not be resolved.",
+         6 => "Couldn't resolve host. The given remote host was not resolved.",
+         7 => "Failed to connect to host.",
+        22 => "HTTP page not retrieved. The requested url was not found or returned another error.",
+        28 => "Operation timeout. The specified time-out period was reached according to the conditions.",
+        35 => "SSL connect error. The SSL handshaking failed.",
+        47 => "Too many redirects. When following redirects, curl hit the maximum amount.",
+        51 => "The peer's SSL certificate or SSH MD5 fingerprint was not OK.",
+        58 => "Problem with the local certificate.",
+        60 => "Peer certificate cannot be authenticated with known CA certificates.",
+        67 => "The user name, password, or similar was not accepted and curl failed to log in.",
+        77 => "Problem with reading the SSL CA cert (path? access rights?).",
+        78 => "The resource referenced in the URL does not exist.",
+       127 => "You requested network access with curl but $system_curl was not found",
+    );
+
+    return '' if (scalar(@params) == 0); ## no parameters provided
+
+    # debug("CURL: %s\r\n",join("\r\n      ",@params)); ## Uncomment only if required as may expose credentials
+
+    ## Use system File::Temp to generate filename...
+    use File::Temp;
+    $tfh = File::Temp->new( TMPDIR => 1,  ## in system temp dir
```

Comment:
So if we're going to just assume `/tmp` then why bother using `File::Temp` in the first place?  FYI on my MacOS `TMPDIR=/var/folders/cq/180yzlm13g5c_66tv8gky5pw0000gn/T/` so there are systems that provide something other than `/tmp` (though MacOS has this as well).

Reply:
> why bother using `File::Temp` in the first place?

Because File::Temp does other stuff too, such as `XXXXXX` substitution and automatic file deletion at scope exit.

> there are systems that provide something other than `/tmp` (though MacOS has this as well).

As long as everyone supports `/tmp` then using `/tmp` avoids the need to escape shell metacharacters in `$ENV{TMPDIR}`, I'm OK with using `TMPDIR => 1` instead of `DIR => '/tmp'` as long as you properly escape special characters in the resulting pathname.

Reply:
I have hard coded it to /tmp/